packages/lesswrong/server/repos/VotesRepo.ts

```diff
@@ -109,4 +116,48 @@ export default class VotesRepo extends AbstractRepo<DbVote> {
       WHERE ARRAY_POSITION("authorIds", $1) IS NOT NULL
     `, [oldUserId, newUserId]);
   }
+
+  // Get votes from recent content by a user,
+  // to use to decide on their rate limit
+  // (note: needs to get the user's own self-upvotes so that
+  // it doesn't skip posts with no other votes)
+  async getVotesOnRecentContent(userId: string): Promise<RecentVoteInfo[]> {
+    const voteFields = `"Votes"._id, "Votes"."userId", "Votes"."power", "Votes"."documentId", "Votes"."collectionName", `
+    const votes = await this.any(`
+      (
+        SELECT ${voteFields} "Posts"."postedAt"
+        FROM "Votes"
+        JOIN "Posts" on "Posts"._id = "Votes"."documentId"
+        WHERE
+          "Votes"."documentId" in (
+            SELECT _id FROM "Posts" 
+            WHERE
+              "Posts"."userId" = '${userId}'
+            ORDER BY "Posts"."postedAt" DESC
+            LIMIT ${RECENT_CONTENT_COUNT}
```

Comment:
I would either hard-code this, or also pass it in as a parameter.  I think you can reference the same parameter value twice in one query (i.e. use `$2` here and also in the other half of the union)

Reply:
Is there any way to get rid of SQL injection risk without resorting to the $2 thing? I think the $1 $2 etc pattern makes things a lot less readable and I'd prefer to have some other way of handling this.